src/Billing/Claim.php

```diff
@@ -44,6 +47,135 @@ class Claim
     public $facilityService;   // via matthew.vita orm work :)
     public $pay_to_provider;   // to be implemented in facility ui
 
+    public function __construct($pid, $encounter_id)
+    {
+        $this->pid = $pid;
+        $this->encounter_id = $encounter_id;
+        $this->encounter = (new EncounterService())->getOneByPidEid($this->pid, $this->encounter_id);
+        $this->getProcsAndDiags($this->pid, $this->encounter_id);
+        $this->copay = $this->getCopay($this->pid, $this->encounter_id);
+        $this->facilityService = new FacilityService();
+        $this->facility = $this->facilityService->getById($this->encounter['facility_id']);
+        $this->pay_to_provider = ''; // will populate from facility someday :)
+        $this->x12_partner = $this->getX12Partner($this->procs[0]['x12_partner_id']);
+        $this->provider = (new UserService())->getUser($this->encounter['provider_id']);
+        $this->billing_facility = empty($this->encounter['billing_facility']) ?
+            $this->facilityService->getPrimaryBillingLocation() :
+            $this->facilityService->getById($this->encounter['billing_facility']);
+        $this->insurance_numbers = $this->getInsuranceNumbers($this->procs[0]['payer_id'], $this->encounter['provider_id']);
+        $this->patient_data = (new PatientService())->findByPid($this->pid);
+        $this->billing_options = $this->getMiscBillingOptions($this->pid, $this->encounter_id);
+        $this->referrer = (new UserService())->getUser($this->getReferrerId());
+        $this->billing_prov_id = (new UserService())->getUser($this->billing_options['provider_id'] ?? null);
+        $this->supervisor = (new UserService())->getUser($this->encounter['supervisor_id']);
+        $this->supervisor_numbers = $this->getInsuranceNumbers($this->procs[0]['payer_id'], $this->encounter['supervisor_id']);
+    }
+
+    public function getProcsAndDiags($pid, $encounter_id)
```

Comment:
for functions that are only using within class, would make them private (I am not sure if that is the case here, though)

Reply:
since this class is called pretty heavily from the X12 Gen class will have to leave as public methods :(